mbedtls/src/cipher/raw/serde.rs

```diff
@@ -87,6 +90,35 @@ impl<Op: Operation> Serialize for Cipher<Op, Traditional, CipherData> {
                 | (CIPHER_ID_3DES, MODE_CFB) => AlgorithmContext::Des3(Bytes(
                     *(cipher_context.cipher_ctx as *const des3_context),
                 )),
+                (CIPHER_ID_AES, MODE_GCM) => {
+                    let gcm_context = *(cipher_context.cipher_ctx as *const gcm_context);
+
+                    let mut inner_ctx = gcm_context.cipher_ctx;
```

Comment:
Instead of manually serializing the inner context, you should just recursively call this logic. Same thing for deserialization.